mesonbuild/mcompile.py

```diff
@@ -15,19 +15,80 @@
 """Entrypoint script for backend agnostic compile."""
 
 import os
-import pathlib
 import shutil
 import sys
 import typing as T
+from pathlib import Path
 
 from . import mlog
 from . import mesonlib
+from . import coredata
 from .mesonlib import MesonException
 
 if T.TYPE_CHECKING:
     import argparse
+    
+def validate_builddir(builddir: Path):
+    if (not builddir.is_dir() or
+        not (builddir / 'meson-private').is_dir() or
+        not (builddir / 'meson-info').is_dir() or
+        not (builddir / 'meson-info' / 'meson-info.json').is_file()):
```

Comment:
Why do you check each subdir? Can't you just check if coredata.dat exists?

Reply:
Haha, fair enough (I've just blindly copy pasted this check from mintro)

Reply:
Right, my 2 remaining nitpickings are not your fault, was already like that, but while you're here it would be nice to cleanup :)

Reply:
I like to leave the code in the better state than I've found it, so all nit-picks are appreciated.

Reply:
Fixed